npm/private/npm_translate_lock.bzl

```diff
@@ -386,12 +387,26 @@ def _gen_npm_imports(lockfile, root_package, attr, registries, default_registry)
                     registry = utils.npm_registry_url(name, registries, default_registry)
                 url = "{0}/{1}".format(registry.removesuffix("/"), tarball)
 
+        if not url:
+            url = utils.npm_registry_download_url(name, version, registries, default_registry)
+
+        _registry = url.split("//", 1)[-1]
+        npm_token = None
+        match_len = 0
+        for (auth_registry, auth_token) in npm_tokens.items():
+            if _registry.startswith(auth_registry) and len(auth_registry) > match_len:
+                npm_token = auth_token
+                match_len = len(auth_registry)
+        npm_auth = ("""
+        npm_auth = "%s",""" % npm_token) if npm_token else ""
```

Comment:
Moved this into `_gen_npm_imports` so that the bzlmod extension can easily share the logic.